Packages/com.unity.inputsystem/InputSystem/Devices/Keyboard.cs

```diff
@@ -311,37 +388,64 @@ public enum Key
     }
 
     /// <summary>
-    /// A keyboard input device.
+    /// Represents a standard, physical PC-type keyboard.
     /// </summary>
     /// <remarks>
-    /// Keyboards allow for both individual button input as well as text input.
+    /// Keyboards allow for both individual button input as well as text input. To receive button
+    /// input, use the individual <see cref="KeyControl"/>-type controls present on the keyboard.
+    /// For example, <see cref="aKey"/>. To receive text input, use the <see cref="onTextInput"/>
+    /// callback.
     ///
-    /// Be aware that identification of keys in the system is layout-agnostic. For example, the
-    /// key referred to as the "a" key is always the key to the right of Caps Lock regardless of
-    /// where the current keyboard layout actually puts the "a" character (if it even has any).
-    /// To find what is actually behind a key according to the current keyboard layout, use
-    /// <see cref="InputControl.displayName"/>, <see cref="KeyControl.shiftDisplayName"/>, and
-    /// <see cref="KeyControl.altDisplayName"/>.
+    /// The naming/identification of keys is agnostic to keyboard layouts. This means that <see cref="aKey"/>,
+    /// for example, will always be the key to the right of <see cref="capsLockKey"/> regardless of where
+    /// the current keyboard language layout puts the "a" character. This also means that having a
+    /// binding to <c>"&lt;Keyboard&gt;/a"</c> on an <see cref="InputAction"/>, for example, will
+    /// bind to the same key regardless of locale -- an important feature, for example, for getting
+    /// stable WASD bindings.
+    ///
+    /// To find what text character (if any) is produced by a key, you can use the key's <see
+    /// cref="InputControl.displayName"/> property. This can also be used in bindings.
+    /// <c>"&lt;Keyboard&gt;/#(a)"</c>, for example, will bind to the key that produces the "a"
+    /// character according to the currently active keyboard layout.
+    ///
+    /// To find out which keyboard layout is currently active, you can use the <see cref="keyboardLayout"/>
+    /// property. Note that keyboard layout names are platform-dependent.
     /// </remarks>
     [InputControlLayout(stateType = typeof(KeyboardState), isGenericTypeOfDevice = true)]
     public class Keyboard : InputDevice, ITextInputReceiver
     {
+        /// <summary>
+        /// Total number of key controls on a keyboard, i.e. the number of controls
+        /// in <see cref="allKeys"/>.
+        /// </summary>
+        /// <value>Total number of key controls.</value>
         public const int KeyCount = (int)Key.OEM5;
 
         /// <summary>
         /// Event that is fired for every single character entered on the keyboard.
         /// </summary>
+        /// <remarks>
+        /// <example>
+        /// <code>
+        /// Keyboard.current.onTextInput +=
+        ///     ch =>
+        ///     {
+        ///         TODO
```

Comment:
No TODO in public docs, please - but i guess in this case it is not so much for us to do, but for the reader of the docs to fill? Then add a comment saying so.

Reply:
Heh, no this is for me. The current script ref docs are still a first pass. Have to admit my approach is not the most structured one. I just go all over the place and repeatedly bang my head against the API doc comments until the result is starting to look decent in the output. So, some stuff here still needs a second pass.